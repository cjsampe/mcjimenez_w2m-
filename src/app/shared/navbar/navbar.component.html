<mat-toolbar color="primary">
  <div class="menu">
    <div>
      <span
        ><strong> {{ "NAVBAR.TITLE_NAVBAR" | transloco }} </strong></span
      >
      <button mat-button routerLink="/home">
        {{ "NAVBAR.HOME" | transloco }}
      </button>
      <button mat-button routerLink="/heroes">
        {{ "HEROES.HEROES" | transloco }}
      </button>
    </div>
    <div></div>
    <div>
      <mat-form-field class="select" appearance="fill">
        <mat-label> {{ "LANG.LANGUAGE" | transloco }}</mat-label>
        <mat-select
          [ngModel]="selectLang"
          (ngModelChange)="selectLanguage($event)"
          name="food"
        >
          <mat-option
            *ngFor="let lenguage of lenguages"
            [value]="lenguage.value"
          >
            {{ lenguage.viewValue }}
          </mat-option>
        </mat-select>
      </mat-form-field>
    </div>
  </div>
</mat-toolbar>
