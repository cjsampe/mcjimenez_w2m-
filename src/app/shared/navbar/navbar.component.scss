body {
    font-family: Roboto, "Helvetica Neue", sans-serif;
    margin: 2em;
    padding: 3em;
    width: 100% !important;
  }
  
  html, body {
    height: 100%;
    width: 100%;
  }
  .menu{
    width: 100%;
    display: flex;
    justify-content: space-between !important;
    margin-top:2em !important;
  }
  .button{
    margin-right: 1em;
    margin-left: 1em;
  }

  mat-toolbar{
    height: 100px !important;
  } 