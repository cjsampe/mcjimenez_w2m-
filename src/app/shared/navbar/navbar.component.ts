import { Component } from '@angular/core';
import { TranslocoService } from '@ngneat/transloco';

interface Lenguages {
  value: string;
  viewValue: string;
}
@Component({
  selector: 'app-navbar',
  templateUrl: './navbar.component.html',
  styleUrls: ['./navbar.component.scss'],
})
export class NavbarComponent {
  selectLang = 'es';

  selectedValue: string;

  lenguages: Lenguages[] = [
    { value: 'en', viewValue:  "LANG.EN" },
    { value: 'es', viewValue: "LANG.ES" },
  ];

  constructor(private translocoService: TranslocoService) {
    this.selectLanguage();
  }

  selectLanguage(LANG: string = this.selectLang) {
    this.translocoService.setActiveLang(LANG);
  }
}
