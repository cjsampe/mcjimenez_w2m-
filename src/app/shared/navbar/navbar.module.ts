import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NavbarComponent } from './navbar.component';
import { MatSlideToggleModule } from '@angular/material/slide-toggle';
import { MatCardModule } from '@angular/material/card';
import { MatIconModule } from '@angular/material/icon';
import { MatToolbarModule } from '@angular/material/toolbar';
import { TranslocoModule } from '@ngneat/transloco';
import { MatButtonModule } from '@angular/material/button';
import { MatSelectModule } from '@angular/material/select';
import { FormsModule } from '@angular/forms';
import { AppRoutingModule } from '../../app-routing.module';




@NgModule({
  declarations: [
    NavbarComponent
  ],
  imports: [
    CommonModule,
    MatCardModule,
    MatSlideToggleModule,
    MatIconModule,
    MatToolbarModule,
    TranslocoModule,
    MatButtonModule,
    MatSelectModule,
    FormsModule,
    AppRoutingModule
  ],
  exports: [
    NavbarComponent
  ]
})
export class NavbarModule { }
