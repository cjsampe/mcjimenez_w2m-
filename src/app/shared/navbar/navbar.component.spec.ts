import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MatToolbarModule } from '@angular/material/toolbar';
import { MatButtonModule } from '@angular/material/button';

import { NavbarComponent } from './navbar.component';
import { TranslocoModule } from '@ngneat/transloco';

describe('NavbarComponent', () => {
  let component: NavbarComponent;
  let fixture: ComponentFixture<NavbarComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [TranslocoModule, MatToolbarModule, MatButtonModule],
      declarations: [NavbarComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(NavbarComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  //create component
  it('should create', () => {
    expect(component).toBeTruthy();
  });

  //snapshot
  it('must be equal to snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  //language
  it('must change language', () => {
    component.selectLanguage('en');
  });
});
