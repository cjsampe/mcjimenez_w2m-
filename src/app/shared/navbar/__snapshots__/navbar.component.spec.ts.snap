// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarComponent must be equal to snapshot 1`] = `
<div
  id="root1"
>
  <mat-toolbar
    class="mat-toolbar mat-primary mat-toolbar-single-row"
    color="primary"
  >
    <div
      class="menu"
    >
      <div>
        <span>
          <strong>
             es.NAVBAR.TITLE_NAVBAR 
          </strong>
        </span>
        <button
          class="mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
          mat-button=""
          routerlink="/home"
        >
          <span
            class="mat-mdc-button-persistent-ripple mdc-button__ripple"
          />
          <span
            class="mdc-button__label"
          >
             es.NAVBAR.HOME 
          </span>
          <span
            class="mat-mdc-focus-indicator"
          />
          <span
            class="mat-ripple mat-mdc-button-ripple"
            matripple=""
          />
          <span
            class="mat-mdc-button-touch-target"
          />
        </button>
        <button
          class="mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
          mat-button=""
          routerlink="/heroes"
        >
          <span
            class="mat-mdc-button-persistent-ripple mdc-button__ripple"
          />
          <span
            class="mdc-button__label"
          >
             es.HEROES.HEROES 
          </span>
          <span
            class="mat-mdc-focus-indicator"
          />
          <span
            class="mat-ripple mat-mdc-button-ripple"
            matripple=""
          />
          <span
            class="mat-mdc-button-touch-target"
          />
        </button>
      </div>
      <div />
      <div>
        <mat-form-field
          appearance="fill"
          class="select"
        >
          <mat-label>
             es.LANG.LANGUAGE
          </mat-label>
          <mat-select
            name="tranlations"
          >
            <mat-option>
               es.LANG.EN 
            </mat-option>
            <mat-option>
               es.LANG.ES 
            </mat-option>
            
          </mat-select>
        </mat-form-field>
      </div>
    </div>
  </mat-toolbar>
</div>
`;
